iceoryx2-ffi/cxx/include/iox2/pending_response.hpp

```diff
@@ -162,77 +204,93 @@ template <ServiceType Service,
 template <typename T, typename>
 inline auto PendingResponse<Service, RequestPayload, RequestHeader, ResponsePayload, ResponseHeader>::user_header()
     -> const T& {
-    IOX_TODO();
+    const void* ptr = nullptr;
+    iox2_pending_response_user_header(&m_handle, &ptr);
+    return *static_cast<const T*>(ptr);
 }
 
 template <ServiceType Service,
           typename RequestPayload,
           typename RequestHeader,
           typename ResponsePayload,
           typename ResponseHeader>
-inline auto PendingResponse<Service, RequestPayload, RequestHeader, ResponsePayload, ResponseHeader>::payload()
-    -> const RequestPayload& {
-    IOX_TODO();
+template <typename T, typename>
+inline auto PendingResponse<Service, RequestPayload, RequestHeader, ResponsePayload, ResponseHeader>::payload() const
+    -> const T& {
+    const void* ptr = nullptr;
+    size_t number_of_elements = 0;
```

Comment:
Please remove `number_of_elements` and provide a `nullptr` since you are not using it.